src/Bicep.Cli/Services/CompilationService.cs

```diff
@@ -107,27 +107,28 @@ public async Task<Compilation> CompileAsync(Uri inputUri, bool skipRestore)
         public async Task<ParamsSemanticModel> CompileParams(string inputPath, bool skipRestore)
         {
             var inputUri = PathHelper.FilePathToFileUrl(inputPath);
+            var configuration = this.configurationManager.GetConfiguration(inputUri);
 
-            if (!fileResolver.TryRead(inputUri, out var fileText, out var failureMessage))
+            var sourceFileGrouping = SourceFileGroupingBuilder.Build(this.fileResolver, this.moduleDispatcher, this.workspace, inputUri, configuration);
+            if (!skipRestore)
             {
-                throw new Exception($"Unable to read file {inputPath}");
+                // module references in the file may be malformed
+                // however we still want to surface as many errors as we can for the module refs that are valid
+                // so we will try to restore modules with valid refs and skip everything else
+                // (the diagnostics will be collected during compilation)
+                if (await moduleDispatcher.RestoreModules(configuration, moduleDispatcher.GetValidModuleReferences(sourceFileGrouping.GetModulesToRestore(), configuration)))
```

Comment:
>RestoreModules [](http://example.com/codeflow?start=43&length=14)

There are no module references in param files today. Are you adding this for a future scenario?

Reply:
Good point - I'm keeping it functionally equivalent to the existing code (which would compile the `.bicep` file, resulting in a restore). Technically if we're just interested in verifying the contract between params and initial bicep file, we may be able to avoid the full compilation. Something to think about for the future.